File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        n = len(nums)
        nums.sort()
        
        for i in range(1, n):
            nums[i] += nums[i - 1]
            
        ans = []
        
        for q in queries:
            ans.append(bisect_right(nums, q))
            
        return ans

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& queries) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        vector<int> prefixSum(n + 1);
        for (int i = 0; i < n; ++i) {
            prefixSum[i + 1] = prefixSum[i] + nums[i];
        }
        int m = queries.size();
        vector<int> ans;
        for (int i = 0; i < m; ++i) {
            ans.push_back(upper_bound(
                prefixSum.begin(), prefixSum.end(), queries[i]) - prefixSum.begin() - 1);
        }
        return ans;
    }
};